### PR TITLE
fix: address https-proxy-agent vulnerability

### PR DESCRIPTION
**Pull request by miiila**
Non-fixed vulnerability is in one of transitive deps, in order to not break customers build, whole train was forked and vuln was patched there.

- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Introduces vuln fix

Appart from changes in this PR, here are the forks updates:
- https://github.com/TooTallNate/node-proxy-agent/compare/master...snyk:fix/https-agent-vuln
- https://github.com/TooTallNate/node-pac-proxy-agent/compare/master...snyk:fix/https-proxy-agent-vuln
- https://github.com/kadler15/node-https-proxy-agent/compare/master...snyk:fix/https-agent-vuln

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,6 +54,7 @@
     "@snyk/cli-interface": "^2.0.3",
     "@snyk/dep-graph": "1.12.0",
     "@snyk/gemfile": "1.2.0",
+    "@snyk/snyk-cocoapods-plugin": "1.0.2",
```
Comment (adrukh) [inline package.json]: emmm intended?

Comment (adrukh) [reply package.json]: of course, it just moved... 🤦‍♂

Comment (miiila) [reply package.json]: https://github.com/snyk/snyk/pull/790/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2L76

Yes ;-)

File: package.json [hunk 2]
```diff
@@ -69,11 +70,10 @@
     "needle": "^2.2.4",
     "opn": "^5.5.0",
     "os-name": "^3.0.0",
-    "proxy-agent": "^3.1.0",
+    "proxy-agent": "git://github.com/snyk/node-proxy-agent.git#fix/https-agent-vuln",
```
Comment (lili2311) [inline package.json]: can you link to the relevant diff here?

Comment (miiila) [reply package.json]: @lili2311 it's in PR description...

TooTallNate/node-proxy-agent@master...snyk:fix/https-agent-vuln
TooTallNate/node-pac-proxy-agent@master...snyk:fix/https-proxy-agent-vuln
kadler15/node-https-proxy-agent@master...snyk:fix/https-agent-vuln

**Pull request merged**